### PR TITLE
Session cache with injected config and network [run-systemtest]

### PR DESCRIPTION
**Pull request by jonmv**
This reapplies session cache with injected config, fixing a problem with chain client setup, and separating out the lowest level representation of "the world" in Slobrok to a reusable component, to handle several graph generations being simultaneously registered as destinations for the same session ID.

### CODE REVIEW

**Comment by jonmv** [comment]
Opted for a multiplexer between message bus and its network. This is set up in an injectable component, and a holder makes sure it is persistent across component graph generations. PTAL @bjorncs .

**Comment by jonmv** [comment]
Hmm, seems there's still something left to do when switching graphs, probably related to the JDisc `Server` setup.

**Comment by jonmv** [comment]
Fixed a bug where non-broadcast sessions wouldn't get messages. Also made sure sessions are deregistered on server-close, which is, AFAICS, the intention. Finally, the last available document protocol is used for decoding incoming messages, which is the same as the current, where self-subscription means the protocol is immediately updated. In fact, this change is better, as the new protocol won't be available very much too early for restart-on-deploy users.

File: container-messagebus/src/main/java/com/yahoo/messagebus/jdisc/test/RemoteClient.java [hunk 1]
```diff
@@ -40,28 +46,22 @@ public Reply awaitReply(int timeout, TimeUnit unit) throws InterruptedException
         return queue.awaitReply(timeout, unit);
     }
 
-    public String slobrokId() {
-        return slobrokId;
+    public SlobroksConfig slobroksConfig() {
+        return TestUtils.configFor(slobrok);
```
Comment (bjorncs) [inline container-messagebus/src/main/java/com/yahoo/messagebus/jdisc/test/RemoteClient.java:50]: Should this method be used "network==true" (production/non-test?) code path in constructor?

File: container-messagebus/src/test/java/com/yahoo/messagebus/jdisc/MbusServerTestCase.java [hunk 2]
```diff
@@ -355,15 +351,13 @@ public String name() {
...
         @Override
         public ResourceReference refer() {
             ++refCount;
-            return new ResourceReference() {
-                @Override
-                public void close() {
-                    --refCount;
-                }
-            };
+            return () -> --refCount;
```
Comment (bjorncs) [inline container-messagebus/src/test/java/com/yahoo/messagebus/jdisc/MbusServerTestCase.java:360]: 😄

File: messagebus/src/main/java/com/yahoo/messagebus/network/NetworkMultiplexer.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+ * time, with all copies potential users of the network interface, but where each message bus-registered session
+ * should belong to a single message bus. This class solves these problems by tracking which sessions are
+ * active in which message bus instance, and by (de)registering only when a session is registered to (no) message
+ * bus instances.
+ *
+ * In time, this should allow us to get rid of the shared-this-and-that in the container, too ...
+ *
+ * @author jonmv
+ */
+public class NetworkMultiplexer implements NetworkOwner {
```
Comment (bjorncs) [inline messagebus/src/main/java/com/yahoo/messagebus/network/NetworkMultiplexer.java:27]: Nice! 🥇

**Review by bjorncs — APPROVED** [review]
👍

File: container-messagebus/src/main/java/com/yahoo/messagebus/jdisc/test/RemoteClient.java (hunk 1 above)
Comment (bjorncs) [reply container-messagebus/src/main/java/com/yahoo/messagebus/jdisc/test/RemoteClient.java:50]: Did not notice the package name...

**Comment by jonmv** [comment]
Unrelated test failure, as far as I can see.

**Pull request merged**